lib/upload.js

```diff
@@ -973,7 +990,7 @@ function openRequest(method, url, options) {
   const headers = options.headers || {}
 
   for (const [name, value] of Object.entries(headers)) {
-    req.setHeader(name, value)
+    req.setHeader(name, BaseUpload.convertToStringHeaderValue(value))
```

Comment:
I don't think we should magically convert non-strings into strings here, especially since this is technically a breaking change. It's the user responsibility to pass proper values in here that match what the documentation mentions.

We can log a warning if the value is not a string and then convert the warning into an error in the next major release.